source/common/router/router.cc

```diff
@@ -525,35 +530,33 @@ void Filter::onResponseTimeout() {
   ENVOY_STREAM_LOG(debug, "upstream timeout", *callbacks_);
   cluster_->stats().upstream_rq_timeout_.inc();
 
-  // It's possible to timeout during a retry backoff delay when we have no upstream request.
-  if (upstream_request_) {
-    if (upstream_request_->upstream_host_) {
-      upstream_request_->upstream_host_->stats().rq_timeout_.inc();
+  ASSERT(upstream_requests_.size() <= 1);
+  if (upstream_requests_.size() == 1) {
+    UpstreamRequest* upstream_request = upstream_requests_.front().get();
```

Comment:
nit: why do we need to grab the raw pointer here? Is to just avoid calling `front()` a bunch. If so I would either just call `front()` or grab a reference and not a pointer for non-null clarity.

Reply:
was just to avoid calling `front()`, will change